examples/rule22_1D_CA.jl

```diff
@@ -1,11 +1,60 @@
 # # One-dimensional cellular automata
-# Agents.jl provides a module (CA1D) to create and plot 1D cellular automata.
 
 using Agents
 using Agents.CA1D
 using AgentsPlots
 
-# ## 1. Define the rule
+# ## 1. Define agent object and stepping function
+
+mutable struct Cell1D <: AbstractAgent
+  id::Int
+  pos::Tuple{Int, Int}
+  status::String
+end
+
+"""
+    build_model(;rules::Dict, ncols=101)
+
+Builds a 1D cellular automaton. `rules` is a dictionary with this format: `Dict("000" => "0")`. `ncols` is the number of cells in a 1D CA.
+"""
+function build_model(;rules::Dict, ncols::Integer=101)
+  nv=(ncols,1)
+  space = GridSpace(nv)
+  properties = Dict(:rules => rules)
+  model = ABM(Cell1D, space; properties = properties, scheduler=by_id)
+  for n in 1:ncols
+    add_agent!(Cell1D(n, (n,1), "0"), (n,1), model)
+  end
+  return model
+end
+
+function ca_step!(model)
+  agentnum = nagents(model)
+  new_status = Array{String}(undef, agentnum)
+  for agid in 1:agentnum
+    agent = model.agents[agid]
+    center = agent.status
+    if agent.id == agentnum
+      # right = "0"  # this is for when there space is not toroidal
+      right =  model.agents[1].status
+      left = model.agents[agent.id-1].status
+    elseif agent.id == 1
+      # left = "0"  # this is for when there space is not toroidal
```

Comment:
When using Literate.jl, comments inside code must start with `##` instead of just 1. OTherwise they are interpreted as markdown, and this ruins all code.